src/Exporter/WebformFormatted/NestedData.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Drupal\campaignion_csv\Exporter\WebformFormatted;
+
+/**
+ * Wrapper for lazy data recursion to lazily load special attributes.
+ */
+class NestedData {
+
+  /**
+   * The data object that is being wrapped.
+   *
+   * @var mixed
+   */
+  protected $data;
+
+  /**
+   * Array of lazy loading callables keyed by the name of the property.
+   *
+   * @var callable[]
+   */
+  protected $overloaded;
+
+  /**
+   * Create a new instance by passing the data object that should be wrapped.
+   *
+   * @param mixed $data
+   *   The data that should be wrapped.
+   * @param array $overloaded
+   *   An array of lazy loading callables for loading special properties.
+   */
+  public function __construct($data, array $overloaded = []) {
+    $this->data = $data;
+    $this->overloaded = $overloaded;
+    foreach (get_class_methods(static::class) as $method) {
+      if (substr($method, 0, 1) != '_') {
+        $this->overloaded[$method] = [$this, $method];
```

Comment:
it's a bit hard to grasp where the "overloaded" callables come from. arguable nice to use, if you know how to...

I tend to see these introspection based coding as too much magic for my taste. makes it harder to understand and navigate code.

but this is more a personal note, no blocking comment... :grin:

Reply:
Do you have any other suggestions?

Reply:
no, not really, at the moment...

Reply:
Ok, should we let it sit for a bit and look whether something else comes to our minds?